lib/app/views/main_page.dart

```diff
@@ -67,9 +74,27 @@ class MainPage extends ConsumerWidget {
     );
     if (deviceNode == null) {
       if (isAndroid) {
+
+        var hasNfcSupport = ref.watch(androidNfcSupportProvider);
+        var isNfcEnabled = ref.watch(androidNfcStateProvider);
         return MessagePage(
           graphic: noKeyImage,
-          message: 'Insert or tap your YubiKey',
+          message: hasNfcSupport
+              ? isNfcEnabled
+                ? const Text('Tap or insert your YubiKey')
+                : Column(
```

Comment:
Instead of changing `message` into a Widget, the `actions` member of `MessagePage` should be used for the button. There's an example of this in `app_failure_page.dart`